### PR TITLE
Cookie sync 341

### PR DESCRIPTION
**Pull request by hhhjort**
Partially implements the solution to #341. Takes care of the original request, but none of the extra features discussed in the thread.

### CODE REVIEW

File: docs/endpoints/cookieSync.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+must supply a JSON object to define the list of bidders that may need to be synced.
+
+```
+{
+    "uuid": "blahblah",
+    "bidders": ["appnexus", "rubicon"]
+}
+```
+
+The `uuid` field is due to be deprecated, as it is no longer needed. It just gets passed through unused into the response currently.
```
Comment (dbemiller) [inline docs/endpoints/cookieSync.md]: If the endpoint works without it, don't document it here. With no documentation, it's unlikely anyone will start using it. If it's mentioned, then some people will. Also: it complicates the docs & wastes time for anyone who just wants to know how to use it properly.

Honestly... Kendall said prebid.js wasn't using it, and Prebid Mobile doesn't use Cookies. If we're going to make breaking changes, sooner is better than later. I think we should just delete it from the endpoint now.

File: pbs_light.go [hunk 2]
```diff
@@ -211,14 +211,37 @@ func (deps *cookieSyncDeps) CookieSync(w http.ResponseWriter, r *http.Request, _
...
+	csReqRaw := map[string]json.RawMessage{}
+	err := json.NewDecoder(r.Body).Decode(&csReqRaw)
 	if err != nil {
 		if glog.V(2) {
 			glog.Infof("Failed to parse /cookie_sync request body: %v", err)
 		}
 		http.Error(w, "JSON parse failed", http.StatusBadRequest)
 		return
 	}
+	if UUID, ok := csReqRaw["uuid"]; ok {
```
Comment (dbemiller) [inline pbs_light.go]: I'm pretty confused... you're parsing this, but the value isn't used anywhere. What's the point here?

Comment (hhhjort) [reply pbs_light.go]: I need to parse it of I am going to pass it through. If we a deprecating it then I can not do it. :) Just grabbing it raw leaves in the quotes around the string.

File: docs/endpoints/cookieSync.md (hunk 1 above)
Comment (hhhjort) [reply docs/endpoints/cookieSync.md]: It is now deprecated completely. :) Bandaid yanked!

File: pbs_light.go (hunk 2 above)
Comment (hhhjort) [reply pbs_light.go]: Oh, you are correct that I never put it in `csReq`. But now I have removed `UUID` from the struct and it is now a one element list. Should I keep the struct (balances against the response struct, and eases expanding the API later) or should I just pull the struct and just work with a bidders list?

File: pbs_light_test.go [hunk 3]
```diff
@@ -113,10 +103,70 @@ func TestCookieSyncHasCookies(t *testing.T) {
...
+
+	if csresp.Status != "no_cookie" {
+		t.Errorf("Expected status = no_cookie; got %s", csresp.Status)
+	}
+
+	if len(csresp.BidderStatus) != 0 {
+		t.Errorf("Expected 0 bidder status rows; got %d", len(csresp.BidderStatus))
+	}
+
+	// Now test a missing bidders returns all syncs
```
Comment (dbemiller) [inline pbs_light_test.go]: You might want to split this into two tests.

Extreme "test-minded" engineers will tell you that each test should be tiny--ideally a single assertion. I've tried that, and think it's a bit overkill... but they're definitely right that short tests which check "a single thing" are easier to debug.

It basically comes down to "when a test fails, how much code do I have to read to understand to figure out what went wrong?" Methods like this double that effort.

Comment (hhhjort) [reply pbs_light_test.go]: I had pondered splitting that myself ... so now done.

File: pbs_light.go [hunk 4]
```diff
@@ -184,12 +184,10 @@ func getAccountMetrics(id string) *AccountMetrics {
 }
 
 type cookieSyncRequest struct {
-	UUID    string   `json:"uuid"`
```
Comment (mkendall07) [inline pbs_light.go]: extra fields are ignored right? in other words, legacy requests with UUID continue to work

Comment (hhhjort) [reply pbs_light.go]: That is correct

**Pull request merged**